### PR TITLE
Fix a bug with diagram property view referencing association

### PR DESCRIPTION
**Pull request by gayathrir11**
#### What type of PR is this?

Bug fix

#### What does this PR do / why is it needed ?

Related to finos/legend-studio#1593

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
yes

### CODE REVIEW

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java [hunk 1]
```diff
@@ -14,13 +14,40 @@
...
 import org.finos.legend.engine.protocol.pure.v1.model.SourceInformation;
 
+import java.io.IOException;
+
+@JsonDeserialize(using = PropertyPointer.PropertyPointerDeserializer.class)
 public class PropertyPointer
 {
-    @JsonProperty(value = "class")
-    public String _class;
+    public String propertyOwner;
```
Comment (akphi) [inline legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:31]: let's do `owner` instead so we save some bytes

**Review by akphi — CHANGES_REQUESTED** [review]

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java [hunk 2]
```diff
@@ -14,13 +14,40 @@
...
+        public PropertyPointer deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException
+        {
+            ObjectMapper om = PureProtocolObjectMapperFactory.getNewObjectMapper();
+            JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+            JsonNode _class = node.get("class");
+            JsonNode property = node.get("property");
+            JsonNode sourceInformation = node.get("sourceInformation");
+            JsonNode propertyOwner = node.get("propertyOwner");
+            PropertyPointer propertyPointer = new PropertyPointer();
+            propertyPointer.propertyOwner = _class != null ? _class.asText() : propertyOwner != null ? propertyOwner.asText() : null;
```
Comment (akphi) [inline legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:47]: you should reorder the ternary operation slightly, if `owner` is present, even if class is specified, we don't care.

Comment (akphi) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:47]: Also, I don't see any test for this, could we add one?

**Review by akphi — CHANGES_REQUESTED** [review]

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java (hunk 2 above)
Comment (akphi) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:47]: Also, let's add a test for this

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java [hunk 3]
```diff
@@ -14,13 +14,40 @@
...
+@JsonDeserialize(using = PropertyPointer.PropertyPointerDeserializer.class)
 public class PropertyPointer
 {
-    @JsonProperty(value = "class")
-    public String _class;
+    public String owner;
     public String property;
     public SourceInformation sourceInformation;
+
+    public static class PropertyPointerDeserializer extends JsonDeserializer<PropertyPointer>
```
Comment (akphi) [inline legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:35]: Do we have a test for the old protocol? I want to ensure even if we send in the old protocol, your custom deserializer works

**Review by akphi — CHANGES_REQUESTED** [review]

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java (hunk 3 above)
Comment (gayathrir11) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:35]: I have added a test where we assert on building new protocol from the json with fields from old protocol. Would that be sufficient?

**Review by akphi — APPROVED** [review]

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java (hunk 3 above)
Comment (akphi) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/mapping/PropertyPointer.java:35]: yep. Thanks!

**Comment by finos-admin** [comment]
This PR is stale because it has been open for 30 days with no activity. Please remove stale label or add any comment to keep this open. Otherwise this will be closed in 5 days.

**Comment by finos-admin** [comment]
This PR was closed because it has been inactive for 35 days. Please re-open if this PR is still relevant.

**Pull request closed without merging**